src/query.rs

```diff
@@ -428,7 +429,11 @@ impl Query {
             .iter()
             .position(|txid| txid == tx_hash)
             .chain_err(|| format!("missing txid {}", tx_hash))?;
-        let (branch, _root) = create_merkle_branch_and_root(txids, pos);
+        let tx_nodes: Vec<TxMerkleNode> = txids
+            .into_iter()
+            .map(|txid| TxMerkleNode::from_inner(txid.into_inner()))
+            .collect();
```

Comment:
I don't like this allocation, but removing it is not trivial. I might look at it later and see if I can optimize it.

Reply:
Well, I think it will be nice to have direction conversion functions for such cases which are optimized by performance – and make them part of the hash type system in `rust-bitcoin`

Reply:
I was thinking more in terms of replacing vecs with iterators, but not really sure how applicable it is in this case. I managed to refactor those in the past, so I might try again. :)

Reply:
It seems that have to be a PR to rust-bitcoin though...